helper/genericdrive.h

```diff
@@ -17,30 +17,25 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-#ifndef RESTOREJOB_H
-#define RESTOREJOB_H
+#ifndef GENERICDRIVE_H
+#define GENERICDRIVE_H
 
 #include <QObject>
-#include <QProcess>
-#include <QTextStream>
+#include <QString>
 
-class RestoreJob : public QObject
-{
+class GenericDrive : public QObject {
     Q_OBJECT
 public:
-    explicit RestoreJob(const QString &where);
-
-signals:
-
-private slots:
-    void work();
-
-private:
-    QTextStream out { stdout };
-    QTextStream err { stderr };
-
-    QProcess m_diskpart;
-    int m_where;
+    virtual void init() = 0;
+    virtual void write(const void *buffer, std::size_t size) = 0;
+    virtual int getDescriptor() const = 0;
```

Comment:
I think because on Windows I don't want to call `_open_osfhandle` before I finished unmounting the drive. Also it would be tedious to update another variable everywhere (especially on Linux).